### ISSUE TITLE
[Linux/ARM] System.Linq.Parallel.Tests fail

### ISSUE THREAD

**Issue by hseok-oh** [dotnet/corefx#10210]
CoreCLR: Release / Debug build 
CoreFX: Debug build

```
hseok82@hseok82-rbp3:~/test/test_corefx/bin/tests/AnyOS.AnyCPU.Debug/System.Linq.Parallel.Tests/netcoreapp1.0$ ./corerun xunit.console.netcore.exe System.Linq.Parallel.Tests.dll  -xml testResults.xml -notrait Benchmark=true  -notrait category=OuterLoop -notrait category=failing -notrait category=nonlinuxtests
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.
 
Discovering: System.Linq.Parallel.Tests
Discovered:  System.Linq.Parallel.Tests
Starting:    System.Linq.Parallel.Tests
 
Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Unable to cast object of type 'System.Int32' to type 'Castable'.)) (Unable to cast object of type 'System.Int32' to type 'Castable'.) ---> System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'Castable'.
   at System.Linq.ParallelQuery`1.<>c__1`1.<Cast>b__1_0(TSource elem) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/ParallelQuery.cs:line 101
   at System.Linq.Parallel.SelectQueryOperator`2.SelectQueryOperatorEnumerator`1.MoveNext(TOutput& currentElement, TKey& currentKey) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Unary/SelectQueryOperator.cs:line 121
   at System.Linq.Parallel.ForAllOperator`1.ForAllEnumerator`1.MoveNext(TInput& currentElement, Int32& currentKey) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Unary/ForAllOperator.cs:line 160
   at System.Linq.Parallel.ForAllSpoolingTask`2.SpoolingWork() in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/SpoolingTask.cs:line 398
   at System.Linq.Parallel.SpoolingTaskBase.Work() in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/SpoolingTaskBase.cs:line 67
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/QueryTask.cs:line 100
   at System.Linq.Parallel.QueryTask.<>c.<.cctor>b__10_0(Object o) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/QueryTask.cs:line 75
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Linq.Parallel.QueryTaskGroupState.QueryEnd(Boolean userInitiatedDispose) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/QueryTaskGroupState.cs:line 132
   at System.Linq.Parallel.SpoolingTask.SpoolForAll[TInputOutput,TIgnoreKey](QueryTaskGroupState groupState, PartitionedStream`2 partitions, TaskScheduler taskScheduler) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/SpoolingTask.cs:line 168
   at System.Linq.Parallel.DefaultMergeHelper`2.System.Linq.Parallel.IMergeHelper<TInputOutput>.Execute() in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Merging/DefaultMergeHelper.cs:line 111
   at System.Linq.Parallel.MergeExecutor`1.Execute() in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Merging/MergeExecutor.cs:line 103
   at System.Linq.Parallel.MergeExecutor`1.Execute[TKey](PartitionedStream`2 partitions, Boolean ignoreOutput, ParallelMergeOptions options, TaskScheduler taskScheduler, Boolean isOrdered, CancellationState cancellationState, Int32 queryId) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/Merging/MergeExecutor.cs:line 92
   at System.Linq.Parallel.PartitionedStreamMerger`1.Receive[TKey](PartitionedStream`2 partitionedStream) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/PartitionedStreamMerger.cs:line 61
   at System.Linq.Parallel.ForAllOperator`1.WrapPartitionedStream[TKey](PartitionedStream`2 inputStream, IPartitionedStreamRecipient`1 recipient, Boolean preferStriping, QuerySettings settings) in /home/hseok82/rbp3/corefx/src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Unary/ForAllOperator.cs:line 95
Aborted (core dumped)
```

Castable class (5b2ae6a)
```
// src/System.Linq.Parallel/tests/QueryOperators/CastTests.cs
        private class Castable
        {   
            private readonly int _value;

            private Castable(int value)
            {   
                _value = value;
            }

            public int Value { get { return _value; } }

            public static explicit operator Castable(Int32 value)
            {
                return new Castable(value);
            }
        }
```

**Comment by hseok-oh** [comment]
Fail in two test methods:
System.Linq.Parallel.Tests.Cast_Unordered_Assignable_InvalidCastException
System.Linq.Parallel.Tests.Cast_Assignable_InvalidCastException

**Comment by hseok-oh** [comment]
fix by dotnet/coreclr#6500
